nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+
+    async def call_invoke_user(self, interaction: Interaction):
+        """|coro|
+
+        Interprets the given interaction and invokes the callback as a User Command."""
+        # TODO: Look into function arguments being autoconverted and given? Arg typed "Channel" gets filled with the
+        #  channel?
+        # Is this kinda dumb? Yeah, but at this time it can only return one user.
+        user = self._handle_resolved_user(list(interaction.data["resolved"]["users"].values())[0])
+        if interaction.guild and (member := interaction.guild.get_member(user.id)):
+            await self.invoke_user(interaction, member)
+        else:
+            await self.invoke_user(interaction, user)
+
+    async def invoke_message(self, interaction: Interaction, message: Message, **kwargs):
+        """|coro|
+
+        Invokes the callback with the given interaction, message, and kwargs as a Message Command."""
+        if self.cog_parent:
+            await self.callback(self.cog_parent, interaction, message, **kwargs)
+        else:
+            await self.callback(interaction, message, **kwargs)
+
+    async def invoke_user(self, interaction: Interaction, member: Union[Member, User], **kwargs):
+        """|coro|
+
+        Invokes the callback with the given interaction, member/user, and kwargs as a User Command."""
+        if self.cog_parent:
+            await self.callback(self.cog_parent, interaction, member, **kwargs)
+        else:
+            await self.callback(interaction, member, **kwargs)
+
+    def _get_basic_application_payload(self) -> dict:
+        """Bare minimum payload that both Global and Guild commands can use."""
+        payload = super().payload
+        if self.type is not ApplicationCommandType.chat_input and "options" in payload:
+            payload.pop("options")
+        if self.default_permission is not None:
+            payload["default_permission"] = self.default_permission
+        return payload
+
+    @property
+    def payload(self) -> List[dict]:
+        """Returns a list of Discord "Application Command Structure" payloads.
+
+        Returns
+        -------
+        payloads: List[:class:`dict`]
+            Returns a list containing the global command payload, if enabled, and payloads corresponding to every guild
+            ID specified.
+        """
+        # TODO: This always returns a list, should it be "payloads"? Won't override subcommand payload though.
+        partial_payload = self._get_basic_application_payload()
+
+        ret = []
+        if self.is_guild:
+            for guild_id in self.guild_ids:
+                temp = partial_payload.copy()  # This shouldn't need to be a deep copy as guild_id is on the top layer.
+                temp["guild_id"] = guild_id
+                ret.append(temp)
+        if self.is_global:
+            ret.append(partial_payload)
+        return ret
+
+    def get_guild_payload(self, guild_id: int) -> Optional[dict]:
```

Comment:
See previous about typing collections in 3.8. Not sure if this applies here tbh, but wouldn't harm